src/data/polynomial/ring_division.lean

```diff
@@ -665,6 +666,26 @@ lemma root_set_finite (p : T[X])
   (S : Type*) [comm_ring S] [is_domain S] [algebra T S] : (p.root_set S).finite :=
 set.to_finite _
 
+lemma bUnion_roots_finite (S : Type*) [comm_ring S] [is_domain S] (d : ℕ) (C : ℕ) :
```

Comment:
Is it possible to generalize this to `S` an `R`-algebra (instead of `ℤ`-algebra), for `R` a... I don't know, normed something?

In any case case, can you please add a docstring and maybe a couple of lines starting with `--` in the proof to explain the math idea? Thanks!

Reply:
> Is it possible to generalize this to `S` an `R`-algebra (instead of `ℤ`-algebra), for `R` a... I don't know, normed something?

I thought about it, then `R` needs to be (at least) a `semi_normed_ring` and also have an instance that implies that the ball of radius `C` in `R` is finite. I will investigate to see if we have something like that in `mathlib`. Also, the result will have to find a new home since importing files about `normed_ring` into `data.polynomial.ring_division` gives a cyclic import.

Reply:
@YaelDillies do you know if we have something that implies balls are finite sets? Thanks!

Reply:
I am looking at the moment at `[preorder R]` and `[locally_finite_order R]`. It looks it might work but won't be pretty...

Reply:
`locally_finite_order` is a good guess, but no we don't have any fully algebraic class that says balls are finite. @linesthatinterlace wanted one for informatics, though.

Reply:
Well, after much fooling around, I have decided that the best is probably just to prove the following instead: 
```lean
lemma bUnion_roots_finite (R S : Type*) [semiring R] [comm_ring S] [is_domain S]
(m : R →+* S) (d : ℕ) (U : set R) (h : U.finite) :
  (⋃ (f : R[X]) (hf : f.nat_degree ≤ d ∧ ∀ i, (f.coeff i) ∈ U),
    ((f.map m).roots.to_finset : set S)).finite :=
```

Reply:
I have pushed the new version of `bUnion_roots_finite`. It is a bit longer than the previous one and does not make use of `ext_iff_nat_degree_le` in the end. (But I still think that `ext_iff_nat_degree_le` is worth keeping).